Directory.php

```diff
@@ -175,8 +175,25 @@ public function copy($to, $force = Stream\IStream\Touchable::DO_NOT_OVERWRITE)
                 continue;
             }
 
-            $file->open()->copy($_to, $force);
-            $file->close();
+            // This is not possible to do `$file->open()->copy();
+            // $file->close();` because the file will be opened in read and
+            // write mode. In a PHAR for instance, this operation is
+            // forbidden. So a special care must be taken to open file in read
+            // only mode.
+            $handle = null;
+
+            if (true === $file->isFile()) {
+                $handle = new Read($file->getPathname());
+            } elseif (true === $file->isDir()) {
+                $handle = new Directory($file->getPathName());
+            } elseif (true === $file->isLink()) {
+                $handle = new Link\Read($file->getPathName());
+            }
+
+            if (null !== $handle) {
```

Comment:
Because this is not what we want :-p. We want to instanciate `$handle` and then do the same things for every instanciation.